### PR TITLE
In case of maintenance we should add a message to the exception

### PR DESCRIPTION
**Pull request by DeepDiver1975**
Recognized via debugging another issue

@PVince81 @LukasReschke please review - THX

### CODE REVIEW

**Comment by DeepDiver1975** [comment]
Note to self - this plugin requires unit tests ...

**Comment by karlitschek** [comment]
:+1:

**Comment by MorrisJobke** [comment]
A lot better. Tested and works :+1:

before:
```
$ curl -X PROPFIND -u abc:abc http://localhost/master/remote.php/webdav/ | xmllint --format -
<?xml version="1.0" encoding="utf-8"?>
<d:error xmlns:d="DAV:" xmlns:s="http://sabredav.org/ns">
  <s:exception>Sabre\DAV\Exception\ServiceUnavailable</s:exception>
  <s:message/>
</d:error>
$ curl -X PROPFIND -u abc:abc http://localhost/master/remote.php/webdav/ | xmllint --format -
<?xml version="1.0" encoding="utf-8"?>
<d:error xmlns:d="DAV:" xmlns:s="http://sabredav.org/ns">
  <s:exception>Sabre\DAV\Exception\ServiceUnavailable</s:exception>
  <s:message/>
</d:error>
```

after:
```
$ curl -X PROPFIND -u abc:abc http://localhost/master/remote.php/webdav/ | xmllint --format -
<?xml version="1.0" encoding="utf-8"?>
<d:error xmlns:d="DAV:" xmlns:s="http://sabredav.org/ns">
  <s:exception>Sabre\DAV\Exception\ServiceUnavailable</s:exception>
  <s:message>System in single user mode.</s:message>
</d:error>
$ curl -X PROPFIND -u abc:abc http://localhost/master/remote.php/webdav/ | xmllint --format -
<?xml version="1.0" encoding="utf-8"?>
<d:error xmlns:d="DAV:" xmlns:s="http://sabredav.org/ns">
  <s:exception>Sabre\DAV\Exception\ServiceUnavailable</s:exception>
  <s:message>System in maintenance mode.</s:message>
</d:error>
```

File: lib/private/connector/sabre/maintenanceplugin.php [hunk 1]
```diff
@@ -56,19 +58,19 @@ public function initialize(\Sabre\DAV\Server $server) {
 	 * This method is called before any HTTP method and returns http status code 503
 	 * in case the system is in maintenance mode.
 	 *
-	 * @throws \Sabre\DAV\Exception\ServiceUnavailable
+	 * @throws ServiceUnavailable
 	 * @internal param string $method
 	 * @return bool
 	 */
 	public function checkMaintenanceMode() {
 		if (\OC::$server->getSystemConfig()->getValue('singleuser', false)) {
```
Comment (MorrisJobke) [inline lib/private/connector/sabre/maintenanceplugin.php]: Just to clarify: In this way it completely disables WebDAV in singleuser mode also for the admin ;) I guess this was by intention? @DeepDiver1975

Comment (DeepDiver1975) [inline lib/private/connector/sabre/maintenanceplugin.php]: > Just to clarify: In this way it completely disables WebDAV in singleuser mode also for the admin ;) I guess this was by intention? @DeepDiver1975

Yes - form my understanding this is the expected behavior

**Pull request merged**